### ISSUE TITLE
Implement Variable Generation From New Configuration Files

### ISSUE THREAD

**Issue by GalaJoe** [galatea-associates/fuse-test-data-gen#156]
Requires issue #126, #127 be resolved.

Issue Description
--
Generation currently uses configuration in which user and developer settings are mixed. Designs for user and developer facing configuration files should now have been written, #126. The program logic must be amended to work with these two configuration files as opposed to the mixed variant. This issue is the implementation of variable generation methods for each relevant object.

Design
--
**Branch from #127 if implemented, start with this issue if not as it lays the groundwork for using the new configuration files in generation.**

- Add a new abstract method into generatable named variable_generation()
- Implement concrete "variable_generation()" methods in each domain object 
    - General premise of these are the same as fixed with 2 key differences:
        1. Generation is to occur over **all** dependencies for the object in question
        2. For each possible combination of dependency objects, there is a percentage chance that it will generate. This probability is to be found inside of the user-defined configuration under variable_generation_args. 

Documentation Changes
--
Docstrings for the abstract method, general enough such that it is applicable to all concrete implementations of it. If there is any domain-object specific logic required, then define a new docstring for the concrete implementation. 

Test Evidence
-- 

All existing tests must pass. 

Validation in Develop
--
[Proof issue is resolved in Develop. Should be complete before the Issue is closed]

**Comment by GalateaTom** [comment]
Issue is on hold following my post in #fuse-test-data-gen on 19/11.  Waiting to find out if variable generation is required.

**Comment by WilfGala** [comment]
Not relevant to MVP since only applies to deprecated functionality and not needed for Tampa POC archived branch to work, so issue closed.
